src/System.Private.CoreLib/shared/System/SpanHelpers.Byte.cs

```diff
@@ -199,10 +200,22 @@ public static unsafe int IndexOf(ref byte searchSpace, byte value, int length)
             IntPtr index = (IntPtr)0; // Use IntPtr for arithmetic to avoid unnecessary 64->32->64 truncations
             IntPtr nLength = (IntPtr)length;
 
-            if (Vector.IsHardwareAccelerated && length >= Vector<byte>.Count * 2)
+            if (Avx2.IsSupported || Sse2.IsSupported)
             {
-                int unaligned = (int)Unsafe.AsPointer(ref searchSpace) & (Vector<byte>.Count - 1);
-                nLength = (IntPtr)((Vector<byte>.Count - unaligned) & (Vector<byte>.Count - 1));
+                if (length >= Vector128<byte>.Count * 2)
```

Comment:
Why do `Avx2` and `Sse` kick in for 32-bytes but `Vector<T>` kick in for 32-bytes or 64-bytes?

Reply:
Resolved in other PR